STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/src/_az_ulib_dm_blob.c

```diff
@@ -174,70 +174,111 @@ static az_result copy_blob_to_flash(NXD_ADDRESS* ip, CHAR* resource, CHAR* host,
+                                NX_FALSE, /* If true, input_size is ignored. */
+                                NX_NULL, /* "name" */
+                                NX_NULL, /* "password" */
+                                DCF_WAIT_TIME)) == NX_SUCCESS)
       {
-        if((nx_status = nx_web_http_client_request_initialize(&http_client,
-                                  NX_WEB_HTTP_METHOD_GET, /* GET, PUT, DELETE, POST, HEAD */
-                                  resource, /* "resource" (usually includes auth headers)*/
-                                  host, /* "host" */
-                                  0, /* Used by PUT and POST */
-                                  NX_FALSE, /* If true, input_size is ignored. */
-                                  NX_NULL, /* "name" */
-                                  NX_NULL, /* "password" */
+        // add user agent
+        if((nx_status = nx_web_http_client_request_header_add(&http_client, 
+                                  USER_AGENT_NAME, 
+                                  sizeof(USER_AGENT_NAME) - 1, 
+                                  USER_AGENT_VALUE, 
+                                  sizeof(USER_AGENT_VALUE) - 1, 
                                   DCF_WAIT_TIME)) == NX_SUCCESS)
         {
-          if((nx_status = nx_web_http_client_request_header_add(&http_client, 
-                                    USER_AGENT_NAME, 
-                                    sizeof(USER_AGENT_NAME) - 1, 
-                                    USER_AGENT_VALUE, 
-                                    sizeof(USER_AGENT_VALUE) - 1, 
-                                    DCF_WAIT_TIME)) == NX_SUCCESS)
-          {
-            nx_status = nx_web_http_client_request_send(&http_client, DCF_WAIT_TIME);
-            // Download destination pointer, increase after every packet is stored
-            uint32_t total_downloaded_size = 0;
-            while(nx_status == NX_SUCCESS)
+          // send request
+          if((nx_status = nx_web_http_client_request_send(&http_client, 
+                                                          DCF_WAIT_TIME)) == NX_SUCCESS)
             {
+              // grab first chunk
               nx_status = nx_web_http_client_response_body_get(&http_client, &packet_ptr, 500);
-              if((nx_status == NX_SUCCESS) || (nx_status == NX_WEB_HTTP_GET_DONE))
-              {
-                az_span data = az_span_create(packet_ptr->nx_packet_prepend_ptr, 
-                                              packet_ptr->nx_packet_length);
 
-                // calculate destination pointer
-                uint8_t* dest_ptr = (uint8_t*)(address) + total_downloaded_size;
+              // save pointer to first chunk
+              az_span data = az_span_create(packet_ptr->nx_packet_prepend_ptr, 
+                                                  packet_ptr->nx_packet_length);
+
+              // calculate destination pointer
+              uint32_t total_downloaded_size = 0;
```

Comment:
>uint32_t total_downloaded_size = 0; [](http://example.com/codeflow?start=14&length=35)

Consider an inproviment here removing this `total_downloaded_size` variable. In this case, make 
```
uint8_t* dest_ptr = (uint8_t*)(address);
```
and add the data size directly to the `dest_ptr`.
```
dest_ptr += az_span_size(data);
```
Both outside and inside of the `while` loop.